simtools/production_configuration/interpolation_handler.py

```diff
@@ -0,0 +1,197 @@
+
+        self.energy_grids = [
+            (e.data["bin_edges_low"][:-1] + e.data["bin_edges_high"][:-1]) / 2
+            for e in self.evaluators
+        ]
+        self.scaled_events = [
+            scaler.scale_events(return_sum=False) for scaler in self.event_scalers
+        ]
+        self.energy_thresholds = np.array([e.energy_threshold for e in self.evaluators])
+
+        self.data, self.grid_points = self._build_data_array()
+
+    def _build_data_array(self):
+        """
+        Build a data array with interpolated values across all dimensions including energy.
+
+        Returns
+        -------
+        np.ndarray
+            The data array with interpolated values.
+        np.ndarray
+            The corresponding grid points.
+        """
+        # Flatten the energy grid and other dimensions into a combined array
+        flat_data_list = []
+        flat_grid_points = []
+
+        for e, energy_grid, scaled_events in zip(
+            self.evaluators, self.energy_grids, self.scaled_events
+        ):
+            az = np.full(len(energy_grid), e.grid_point[1].to(u.deg).value)
+            zen = np.full(len(energy_grid), e.grid_point[2].to(u.deg).value)
+            nsb = np.full(len(energy_grid), e.grid_point[3])
+            offset = np.full(len(energy_grid), e.grid_point[4].to(u.deg).value)
+
+            # Combine grid points and data
+            grid_points = np.column_stack([energy_grid.to(u.TeV).value, az, zen, nsb, offset])
+            flat_grid_points.append(grid_points)
+            flat_data_list.append(scaled_events)
+
+        # Flatten the list and convert to numpy arrays
+        flat_grid_points = np.vstack(flat_grid_points)
+        flat_data = np.hstack(flat_data_list)
+
+        # Sort the grid points and corresponding data by energy
+        sorted_indices = np.argsort(flat_grid_points[:, 0])
+        sorted_grid_points = flat_grid_points[sorted_indices]
+        sorted_data = flat_data[sorted_indices]
+
+        return sorted_data, sorted_grid_points
+
+    def _remove_flat_dimensions(self, grid_points):
+        """Identify and remove flat dimensions (dimensions with no variance)."""
+        variance = np.var(grid_points, axis=0)
+        non_flat_mask = variance > 1e-6  # Threshold for determining flatness
+        reduced_grid_points = grid_points[:, non_flat_mask]
+        return reduced_grid_points, non_flat_mask
+
+    def interpolate(self, query_points: np.ndarray) -> np.ndarray:
+        """
+        Interpolate the number of simulated events given query points.
+
+        Parameters
+        ----------
+        query_points : np.ndarray
+            Array of query points with shape (n, 5), where n is the number of points,
+            and 5 represents (energy, azimuth, zenith, nsb, offset).
+
+        Returns
+        -------
+        np.ndarray
+            Interpolated values at the query points.
+        """
+        # Remove flat dimensions from the normalized grid and query points
```

Comment:
No need for this comment. `remove_flat_dimensions` is a good function name and explains its functionality.

Reply:
fixed